File: store-admin/src/app/pages/auth/auth.service.ts
import { Injectable } from '@angular/core';
import {BaseService} from "../../services/base.service";
import {AuthRequest, RegisterRequest} from "./auth.model";

@Injectable({
  providedIn: 'root'
})
export class AuthService{
  private readonly apiUrl: string = "/api/v1/auth"
  constructor(private base : BaseService) {
  }
  login(request:AuthRequest){
    return this.base.post(this.apiUrl+"/login",request);
  }
  activeAccount(token:string){
    return this.base.get(this.apiUrl+"/verification/"+token);
  }
  generateActiveToken(username:string){
    return this.base.get(this.apiUrl+"/generation-active-token/"+username)
  }
  forgotPassword(request:AuthRequest){
    return this.base.post(this.apiUrl+"/forgot-password",request);
  }
  activePassword(token : string){
    return this.base.get(this.apiUrl + "/forgot-password/"+token);
  }
  generateNewToken(token : string){
    return this.base.get(this.apiUrl + "/forgot-password/new-token/"+token);
  }
  logout(){
    return this.base.get(this.apiUrl + "/logout");
  }
  refreshToken(token : string){
    return this.base.get(this.apiUrl+"/refresh-token/"+token);
  }
  register(request: RegisterRequest){
    return this.base.post(this.apiUrl+"/register",request);
  }

}

File: application/src/main/java/com/thd/store/service/AuthService.java
package com.thd.store.service;

import com.thd.store.dto.auth.AuthRequest;
import com.thd.store.dto.auth.ForgotPasswordRequest;
import com.thd.store.dto.auth.OAuthRequest;
import com.thd.store.dto.auth.RegisterRequest;
import com.thd.store.dto.BaseResponse;

import java.io.IOException;
import java.security.GeneralSecurityException;

/**
 * @author DatNuclear 16/01/2024
 * @project store-movie
 */
public interface AuthService {
    BaseResponse login(AuthRequest request);
    BaseResponse register(RegisterRequest request);
    BaseResponse verifyAccount(String token);
    BaseResponse generateActiveToken(String username);
    BaseResponse forgotPassword(ForgotPasswordRequest request);
    BaseResponse activeNewPassword(String token);
    BaseResponse generateNewToken(String token);
    BaseResponse refreshToken(String token);
    BaseResponse loginGoogle(OAuthRequest request) throws GeneralSecurityException, IOException;
}
